### PR TITLE
Integrate logic of game

### PR DESCRIPTION
**Pull request by JessickaMoya**

### CODE REVIEW

File: app/src/main/java/src/utilities/Menu.java [hunk 1]
```diff
@@ -24,10 +24,10 @@
...
      */
     public void showGameMenu() {
-        gameBoard = new GameBoard();
-        game = new Game();
-        randomShape = new RandomShape();
-        shape = randomShape.getShape(randomNumberGenerator(), INITIAL_POSX_SHAPE, INITIAL_POSY_SHAPE);
+//        gameBoard = new GameBoard();
+//        game = new Game();
+//        randomShape = new RandomShape();
+//        shape = randomShape.getShape(randomNumberGenerator(), INITIAL_POSX_SHAPE, INITIAL_POSY_SHAPE);
```
Comment (jpgonzales27) [inline app/src/main/java/src/utilities/Menu.java]: these commented lines should be removed

File: app/src/main/java/src/utilities/Menu.java [hunk 2]
```diff
@@ -102,10 +102,30 @@ public void showDifficultyMenu(final Game game, final GameBoard gameBoard, final
     /**
      * Prints the options for movement menu
      */
-    public void showMovementMenu(final Game game, final GameBoard gameBoard, final Shape shape) {
+    public void showMovementMenu() {
+//        gameBoard = new GameBoard();
+//        game = new Game();
+//        randomShape = new RandomShape();
+//        shape = randomShape.getShape(randomNumberGenerator(), INITIAL_POSX_SHAPE, INITIAL_POSY_SHAPE);
+//        gameBoard.setGameBoardArray(shape);
```
Comment (jpgonzales27) [inline app/src/main/java/src/utilities/Menu.java]: these commented lines should be removed

File: app/src/main/java/src/utilities/Menu.java [hunk 3]
```diff
@@ -102,10 +102,30 @@ public void showDifficultyMenu(final Game game, final GameBoard gameBoard, final
...
+        gameBoard = new GameBoard();
+//        boolean[][] joelXD = {{true, true, true, true, true, true}};
+//        gameBoard.setGameBoardArray(0, 6, joelXD);
```
Comment (jpgonzales27) [inline app/src/main/java/src/utilities/Menu.java]: these commented lines should be removed

File: app/src/main/java/src/utilities/Menu.java [hunk 4]
```diff
@@ -102,10 +102,30 @@ public void showDifficultyMenu(final Game game, final GameBoard gameBoard, final
     /**
      * Prints the options for movement menu
      */
-    public void showMovementMenu(final Game game, final GameBoard gameBoard, final Shape shape) {
+    public void showMovementMenu() {
+//        gameBoard = new GameBoard();
```
Comment (JRojas999) [inline app/src/main/java/src/utilities/Menu.java]: Commented Code has to delete

File: app/src/main/java/src/utilities/Menu.java (hunk 1 above)
Comment (JessickaMoya) [reply app/src/main/java/src/utilities/Menu.java]: Thanks, I will erase them.

File: app/src/main/java/src/utilities/Menu.java (hunk 2 above)
Comment (JessickaMoya) [reply app/src/main/java/src/utilities/Menu.java]: Thanks, I will erase them.

File: app/src/main/java/src/utilities/Menu.java (hunk 3 above)
Comment (JessickaMoya) [reply app/src/main/java/src/utilities/Menu.java]: Thanks, I will erase them.

File: app/src/main/java/src/utilities/Menu.java [hunk 5]
```diff
@@ -116,23 +136,38 @@ public void showMovementMenu(final Game game, final GameBoard gameBoard, final S
...
+                        shape.moveDown();
+                    }
                     break;
                 case OPTION_2:
-                    System.out.println("Moved to left");
-                    shape.moveLeft();
-                    shape.moveDown();
+                    if (game.checkCollision(shape, gameBoard, 2 + 1)) {
+                        collision = true;
+                        gameBoard.setGameBoardArray(shape.getxPosition(), shape.getyPosition(), shape.getContainer());
```
Comment (Raymundogs) [inline app/src/main/java/src/utilities/Menu.java]: This method should receive a shape instead of an array

Comment (JessickaMoya) [reply app/src/main/java/src/utilities/Menu.java]: Thanks, I will change it

**Review by jpgonzales27 — APPROVED** [review]

**Review by Raymundogs — APPROVED** [review]

File: app/src/main/java/src/utilities/Menu.java (hunk 1 above)
Comment (JessickaMoya) [reply app/src/main/java/src/utilities/Menu.java]: It is done

File: app/src/main/java/src/utilities/Menu.java (hunk 2 above)
Comment (JessickaMoya) [reply app/src/main/java/src/utilities/Menu.java]: It is done

File: app/src/main/java/src/utilities/Menu.java (hunk 3 above)
Comment (JessickaMoya) [reply app/src/main/java/src/utilities/Menu.java]: It is done

File: app/src/main/java/src/utilities/Menu.java (hunk 5 above)
Comment (JessickaMoya) [reply app/src/main/java/src/utilities/Menu.java]: It is done

**Review by JRojas999 — APPROVED** [review]

File: app/src/main/java/src/utilities/Menu.java (hunk 1 above)
Comment (jpgonzales27) [reply app/src/main/java/src/utilities/Menu.java]: ok jessi, good job

File: app/src/main/java/src/utilities/Menu.java (hunk 2 above)
Comment (jpgonzales27) [reply app/src/main/java/src/utilities/Menu.java]: ok jessi, good job

File: app/src/main/java/src/utilities/Menu.java (hunk 3 above)
Comment (jpgonzales27) [reply app/src/main/java/src/utilities/Menu.java]: ok jessi, good job

File: app/src/main/java/src/utilities/Menu.java (hunk 5 above)
Comment (jpgonzales27) [reply app/src/main/java/src/utilities/Menu.java]: ok jessi, good job

**Pull request merged**